File: articles/history-of-algorithms/c/euclidean_algorithm.c
#include <stdio.h>
#include <stdlib.h>

int gcd_algorithm(int a, int b) {
    while (b != 0) {
        int r = a % b;
        a = b;
        b = r;
    }
    return abs(a);
}

int main(void) {
    printf("gcd=%d\n", gcd_algorithm(252, 105));
    return 0;
}

File: articles/history-of-algorithms/rust/euclidean_algorithm.rs
fn gcd_algorithm(mut a: i64, mut b: i64) -> i64 {
    while b != 0 {
        let r = a % b;
        a = b;
        b = r;
    }
    a.abs()
}

fn main() {
    println!("gcd={}", gcd_algorithm(252, 105));
}
